### PR TITLE
Add time display to viewer controls

### LINKED ISSUE #150: Add time to videos
When I watch a video, I have a red bar for the progress. However I would like to see the timeline. Normally a 0:00 on the left and a maximum time the video will play on the right and then like 1:30 in between which should be in the middle when the video is 3:00 long.

There is more nice to have stuff, but thats kind of a minimum ...

### PR DESCRIPTION
**Pull request by mmattel**
Based on the work of @tomschulze on original PR #223 and #224 (both closed).
Fixes #150 (suggestion from @hodyroff)
Tested on my dev environment and works as described

This adds the time to the video controls, pls see screenshot:

image

Original comment from @tomschulze:

A few remarks:

- I wasn't sure about your styling and I didnt wanna mess with your styles, so I used the `viewer__control__count` css class to style the display for test purposes (which disappears on small screens)
- `totalDisplayTime` is computed because it doesn't change during the video
- I put `getDisplayTime()` in helper.js since it seemed more like a utility function to me
- the function `showTimeInfo()` aggregates the displayed string

### CODE REVIEW

**Comment by phil-davis** [comment]
The code looks believable, but I am not the one to review JS.

File: src/scripts/helper.js [hunk 1]
```diff
@@ -18,6 +18,23 @@ const helper = {
 
 		closeViewer () {
 			this.$router.push('/');
+		},
+
+		getDisplayTime (time) {
+			if (time === 0) return "00:00";
```
Comment (DeepDiver1975) [inline src/scripts/helper.js:24]: I'm a hugh fan of putting each statement into their own line

```if (time === 0) {
    return "00:00";
}

File: src/scripts/helper.js [hunk 2]
```diff
@@ -18,6 +18,23 @@ const helper = {
...
+
+			// when switching between previews, currentTime is not rounded somehow
+			time = Math.round(time);
+
+			let hour = Math.floor(time / 3600);
+			let minute = Math.floor(time % 3600 / 60);
+			let second = time % 60;
+
+			let times = [minute, second];
+			if ( hour > 0 ) times = [hour, minute, second];
```
Comment (DeepDiver1975) [inline src/scripts/helper.js:34]: same as above

File: src/scripts/helper.js [hunk 3]
```diff
@@ -18,6 +18,23 @@ const helper = {
...
+			times = times.map( el => el < 10 ? (el+"").padStart(2, "0") : el);
```
Comment (DeepDiver1975) [inline src/scripts/helper.js:35]: I strongly believe there are ready to use functions which are doing this - oc10 has moment.js already loaded ... please have a look

**Comment by mmattel** [comment]
@DeepDiver1975 fixed all your comments 😃

**Comment by mmattel** [comment]
@DeepDiver1975 you are challenging me 😄 
please take a look on the update, thanks

**Comment by DeepDiver1975** [comment]
24 / 30 - nobody will understand this ....

Screenshot from 2020-04-14 10-42-39

**Comment by mmattel** [comment]
Hello @DeepDiver1975 
First of all, thanks for reviewing and your patience.
When the video plays, you see the left part of the number block counting, while the right part shows the total time. Depending on the length, you see `ss / ss` or `mm:ss / mm:ss` or `hh:mm:ss / hh:mm:ss`. What is your suggestion of presentation?

**Comment by DeepDiver1975** [comment]
> What is your suggestion of presentation?

At least 00:24 / 00:30 I'd say.

**Comment by mmattel** [comment]
OK, will try to have fixed minutes and optional hours as suggested

**Comment by mmattel** [comment]
Hello @DeepDiver1975 implemented your suggestion

**Comment by mmattel** [comment]
@DeepDiver1975 may I ask for a review - would like to get this finalized 😃

**Comment by mmattel** [comment]
Just seen that recently upgrading easytext from another PR we had a conflict in packages.json to be solved - done.

@deepdiver please review so we can merge it - many thanks :blush:

**Pull request merged**